### PR TITLE
librdkafka: add version 1.9.1 and remove older versions

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **librdkafka/***

- add version 1.9.1
  - add `curl` options because librdkafka >=1.9.0 requires libcurl when `WITH_CURL` is ON.
- remove version 1.3.0, 1.4.0, 1.5.2
  - remove patches of these versions too.
- update dependencies
- sort versions
- use `conan.tools.build.cross_building`

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@toge probably the project is using CMake with `find_package`, but without `CONFIG`, so the generator `cmake_find_package_multi` will not work on that case.

**Comment by toge** [comment]
@uilianries 
I totally forgot it.
I will try to fix.
Thank you a lot!

**Review by uilianries — APPROVED** [review]
Thank you @toge !!!

**Review by prince-chrismc — APPROVED** [review]

File: recipes/librdkafka/all/conanfile.py [hunk 1]
```diff
@@ -34,9 +34,10 @@ class LibrdkafkaConan(ConanFile):
         "plugins": False,
         "ssl": False,
         "sasl": False,
+        "curl": False,
```
Comment (prince-chrismc) [inline recipes/librdkafka/all/conanfile.py:37]: too bad these dont follow the convention/rule in the docs

File: recipes/librdkafka/all/conanfile.py [hunk 2]
```diff
@@ -34,9 +34,10 @@ class LibrdkafkaConan(ConanFile):
...
     }
 
-    generators = "cmake", "cmake_find_package", "pkg_config"
+    generators = "cmake", "cmake_find_package_multi", "pkg_config"
```
Comment (SpaceIm) [inline recipes/librdkafka/all/conanfile.py:40]: Please revert this change.

File: recipes/librdkafka/all/patches/0001-Change-library-names-1-9-1.patch [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+index 7f3dd0f..89a21e7 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -67,7 +67,7 @@ endif()
+ # }
+ 
+ # ZSTD {
+-find_package(ZSTD QUIET)
++find_package(zstd QUIET CONFIG)
+ if(ZSTD_FOUND)
```
Comment (SpaceIm) [inline recipes/librdkafka/all/patches/0001-Change-library-names-1-9-1.patch:11]: add `set(ZSTD_FOUND TRUE)`.

File: recipes/librdkafka/all/patches/0002-Change-library-targets-and-result-variables-1-9-1.patch [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -207,9 +207,7 @@ if(WITH_ZLIB)
+ endif()
+ 
+ if(WITH_ZSTD)
+-  target_link_libraries(rdkafka PRIVATE ${ZSTD_LIBRARY})
+-  target_include_directories(rdkafka PRIVATE ${ZSTD_INCLUDE_DIR})
+-  message(STATUS "Found ZSTD: ${ZSTD_LIBRARY}")
++  target_link_libraries(rdkafka PRIVATE zstd::zstd)
```
Comment (SpaceIm) [inline recipes/librdkafka/all/patches/0002-Change-library-targets-and-result-variables-1-9-1.patch:12]: will break with `CMakeDeps`
